### PR TITLE
Integrate ChainerMN to Chainer repository

### PR DESCRIPTION
**Pull request by keisukefukuda**
This PR integrates ChainerMN, an add-on package to support multinode training.
Background:
 - Distributed deep learning is getting more popular, thus Chainer should natively support it
 - As both of Chainer and ChainerMN are continuously updated, it is getting harder to support all version combinations of the softwares.

What this PR does:
 - Move ChainerMN to Chainer's repository and synchronize the release cycle.
 - Integrate `.travis.yml` setup

What this PR doesn't 
 - It does not change the library usage. The `chainermn` package remains as is and existing ChainerMN programs work without modification.
 - It doesn't am to integrate various components of Chainer and ChainerMN. There are several duplicate building blocks between Chainer and ChainerMN, but we focus on integrating the repositories and do not step further for now.

Major TODOs:
 - Integrate `.travis.yml` 
 - Integrate documents.
 - Check coding styles of both projects.
 - Improve error messages when `mpi4py` is not installed

### CODE REVIEW

**Comment by kmaehashi** [comment]
In addition, ChainerMN tests should be excluded from chainer-test execution.
https://github.com/chainer/chainer-test/blob/4afcfb73bee19cd4fbd4daff4fb7f872870a0ce6/test.sh#L38

**Comment by keisukefukuda** [comment]
@kmaehashi Please let us talk offline later on the test issue.

File: chainermn/__init__.py [hunk 1]
```diff
@@ -20,4 +20,4 @@
 
 global_except_hook._add_hook_if_enabled()
 
-__version__ = pkg_resources.get_distribution('chainermn').version
+__version__ = pkg_resources.get_distribution('chainer').version
```
Comment (kmaehashi) [inline chainermn/__init__.py]: Please use `chainer.__version__` instead. (c.f. #3628)

Comment (keisukefukuda) [reply chainermn/__init__.py]: I already found the mistake and fixed it! thanks!

Comment (kmaehashi) [reply chainermn/__init__.py]: Ah, I meant `pkg_resources.get_distribution ` should be avoided. It's totally fine to do this in separate PR.

**Comment by kmaehashi** [comment]
> In addition, ChainerMN tests should be excluded from chainer-test execution. (I'll do this later)

https://github.com/chainer/chainer-test/pull/448

**Comment by keisukefukuda** [comment]
This PR is based on ChainerMN 
f66d35e56810d7e67c509ffb05db3e3a816e6e7b

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM!

File: chainermn/__init__.py (hunk 1 above)
Comment (kmaehashi) [reply chainermn/__init__.py]: I did it in https://github.com/chainer/chainer/pull/5298

**Pull request merged**